store/src/java/com/zimbra/cs/util/ProxyConfGen.java

```diff
@@ -1106,6 +1107,107 @@ public void update() throws ServiceException {
     }
 }
 
+class OnlyOfficeDocServiceServersVar extends ProxyConfVar {
+
+    private final String defaultHost = "zimbra.com";
+
+    public OnlyOfficeDocServiceServersVar() {
+
+        super("web.upstream.onlyoffice.docservice", null, null, ProxyConfValueType.STRING, ProxyConfOverride.CUSTOM,
+                "List of upstream HTTPS servers towards docservice port used by Web Proxy ");
+    }
+
+    @Override
+    public void update() throws ServiceException {
+        StringBuffer sb = new StringBuffer();
+        // check if zimbraDocumentEditingHost has been set
+        // if it is set, it means onlyoffice is to be deployed on a single
+        // server
+        String onlyofficeServer = mProv.getConfig().getDocumentEditingHost();
+        if (onlyofficeServer != null && onlyofficeServer.trim().length() > 0 && !defaultHost.equals(onlyofficeServer)) {
+            Server server = mProv.get(Key.ServerBy.name, onlyofficeServer);
+            sb.append(generateUpstreamBlock(server.getId(), onlyofficeServer));
+            mLog.debug("Added docservice upstream for single onlyoffice server : " + onlyofficeServer);
+        } else {
+            List<Server> mailclientservers = mProv.getAllMailClientServers();
```

Comment:
Broke use case:
1.Scenario where only few of the mailbox servers have running onlyoffice.

Reply:
This isn't handled over here. The 2 scenarios that are handled here are : 
1. onlyoffice on a single server.
2. onlyoffice on each mailbox server

Reply:
@sreejan-chowdhury we need to handle this as per discussion with Product,

Reply:
Changes have been done to handle the scenario too.